src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -165,8 +168,18 @@ else if(item == SCContent.DISGUISE_MODULE.get())
 				ItemStack module = getDisguiseModule();
 
 				if(!module.isEmpty()) //drop the old module as to not override it with the new one
+				{
 					Block.spawnAsEntity(world, getPosition(), module);
 
+					List<ItemStack> blocks = ((ModuleItem)module.getItem()).getAddons(module.getTag());
+					if(blocks.size() > 0)
+					{
+						BlockState state = Block.getBlockFromItem(blocks.get(0).getItem()).getDefaultState();
+						if(state == world.getBlockState(pos))

```

Comment:
This will fail if e.g. an anvil is not rotated the same way as it is in its default state. Check whether the blocks are the same, instead of the state.

Reply:
Good point, I didn't think of that.